agents/meinberg_m1000/meinberg_m1000_agent.py

```diff
@@ -0,0 +1,482 @@
+            message = self._build_message(interval, result, read_time)
+
+            # Update lastGet time
+            for mib in self.mib_timings:
+                if mib['oid'] in get_list:
+                    mib['lastGet'] = read_time
+
+            self.log.debug("{msg}", msg=message)
+            session.app.publish_to_feed('m1000', message)
+            session.data = self.get_cache()
+
+
+class MeinbergM1000Agent:
+    """Monitor the Meinberg LANTIME M1000 timing system via SNMP.
+
+    Parameters
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    address : str
+        Address of the M1000.
+    port : int
+        SNMP port to issue GETs to, default to 161.
+
+    Attributes
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    is_streaming : bool
+        Tracks whether or not the agent is actively issuing SNMP GET commands
+        to the M1000. Setting to false stops sending commands.
+    log : txaio.tx.Logger
+        txaio logger object, created by the OCSAgent
+
+    """
+    def __init__(self, agent, address, port=161):
+        self.agent = agent
+        self.is_streaming = False
+        self.log = self.agent.log
+
+        self.meinberg = MeinbergSNMP(address, port)
+
+        agg_params = {
+            'frame_length': 10*60  # [sec]
+        }
+        self.agent.register_feed('m1000',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1)
+
+    @inlineCallbacks
+    def start_acq(self, session, params=None):
+        """start_acq(params=None)
+
+        OCS Process for fetching values from the M1000 via SNMP.
+
+        The session.data object stores each unique OID with its latest status,
+        decoded value, and the last time the value was retrived. This will look like
+        the example here::
+
+            >>> session.data
+            {"mbgLtNgPtpPortState_1":
+                {"status": 3,
+                 "description": "disabled",
+                 "lastGet": 1598543397.689727},
+            "mbgLtNgNtpCurrentState_0":
+                {"status": 1,
+                 "description": "not synchronized",
+                 "lastGet": 1598543363.289597},
+            "mbgLtNgRefclockState_1":
+                {"status": 2,
+                 "description": "not synchronized",
+                 "lastGet": 1598543359.6326838},
+            "mbgLtNgSysPsStatus_1":
+                {"status": 2,
+                 "description": "up",
+                 "lastGet": 1598543359.6326838},
+            "mbgLtNgSysPsStatus_2":
+                {"status": 2,
+                 "description": "up",
+                 "lastGet": 1598543359.6326838},
+            "mbgLtNgEthPortLinkState_1":
+                {"status": 1,
+                 "description": "up",
+                 "lastGet": 1598543359.6326838}}
```

Comment:
What's the format of LeapSecondDate?  Since you tell me all the other ones...

Reply:
Right, that's actually a str, which differed from other "strings" which were stored as bytes within the returned ObjectType. The types within ObjectType were pretty awkward to handle and I'm mostly empirically figuring out what works. I was initially assuming all strings were initially `bytes`, but now am letting `str` through uncaught as well. I added LeapSecondDate to the docstring. Good catch.